com.unity.multiplayer.mlapi/Runtime/Core/NetworkBehaviourUpdater.cs

```diff
@@ -0,0 +1,93 @@
+using System.Collections.Generic;
+using Unity.Profiling;
+
+namespace MLAPI
+{
+    public class NetworkBehaviourUpdater
```

Comment:
huh, adding this `NetworkBehaviourUpdater` to the MLAPI/Runtime/Core, what's going on here? :)

Reply:
having said that, this brand-new class itself is not being tested even though being introduced and implemented under this PR. I'm really confused.

Reply:
The only new stuff is moving static functions to a non-static class. There's nothing new here, only the way we access the updater (an instance is now stored in NetworkManager instead of accessing it statically)
Since client and server network manager shared the NetworkBehaviour static state, I ran into issues that happened once in a while, but not consistently.

Reply:
nevertheless, that's a separate PR with a good description of what's happening, IMHO.

Reply:
I actually already raised that here, you might have missed it
https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/pull/868#discussion_r642580778

Reply:
I'm not sure if that's enough and again, I still believe it needs to go under a separate PR with proper tests implemented + a good description (which would spawn further questions or clarify them upfront).

Reply:
I can see both points @SamuelBellomo and @MFatihMAR.  Yeah, I'd prefer to have this in a separate refactor PR. 
 As a practical matter, this code I'm pretty certain will migrate out into the snapshot system.  I think on balance, having this in its own file will make that cleaner vs. doing all that work in NetworkBehaviour.  Upshot, I'm ok with this with again the guidance we prefer to do these in separate PRs next time.

Reply:
moved to https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/pull/917